From 42454ea94f45a7436baeb677b1584acbbab3a01c Mon Sep 17 00:00:00 2001
From: 609bob <1850029304@qq.com>
Date: Mon, 26 Oct 2026 19:38:50 +0800
Subject: [PATCH] ssl/bio_ssl.c: reset BIO's data pointer to NULL in ssl_free()

Resolves: https://github.com/openssl/openssl/issues/31388
Fixes: 58964a492275 "Import of old SSLeay release: SSLeay 0.9.0b"
Reviewed-by: Eugene Syromiatnikov <esyr@openssl.org>
Reviewed-by: Nikola Pajkovsky <nikolap@openssl.org>
Merge-date: Wed Oct 28 10:12:45 2026
Merged-from: https://github.com/openssl/openssl/pull/33402
(cherry picked from commit 527b43d3b737a9c2be430e9741599235c3b93755)
---
 ssl/bio_ssl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ssl/bio_ssl.c b/ssl/bio_ssl.c
index ebd9c58..ce58bce 100644
--- a/ssl/bio_ssl.c
+++ b/ssl/bio_ssl.c
@@ -89,6 +89,7 @@ static int ssl_free(BIO *a)
         BIO_set_init(a, 0);
     }
     OPENSSL_free(bs);
+    BIO_set_data(a, NULL);
     return 1;
 }
 
-- 
2.53.0

